src/main/native/pairhmm/JavaData.h

```diff
@@ -3,25 +3,45 @@
 
 #include <vector>
 #include <debug.h>
+#include <exception>
+#include <string>
 #include "pairhmm_common.h"
 
+class JavaException : std::exception {
+  public:
+  const char* classPath;
+  const char* message;
+
+  JavaException(const char* classPath, const char* message)
+    : classPath(classPath),
+      message(message)
+  { }
+};
+
 class JavaData {
  public:
+  JavaData(JNIEnv *env) : m_env(env) {
```

Comment:
Keeping a `JNIEnv` reference inside this class is necessary to perform clean-up in the destructor.